src/io/per/err.rs

```diff
@@ -13,6 +13,7 @@ pub enum Error {
     SizeNotInRange(u64, u64, u64),
     OptFlagsExhausted,
     EndOfStream,
+    InvalidType,
```

Comment:
I'like to see more debug info here. A `InvalidType(String)` in which the `String` contains something like `"Got unexpected tag {:?}"` or so.

Reply:
Yup. Now that we have the `C::TAG`, it is better to change it to that